src/modules/Main.ui.js

```diff
@@ -215,27 +215,23 @@ export default class Main extends Component<Props, State> {
         delay: 500
       })
     })
-    if (Platform.OS === 'android') {
-      Linking.getInitialURL().then(url => {
-        if (url) {
-          this.props.urlRecived(url)
-        }
-        // this.navigate(url);
-      })
-    } else {
-      Linking.addEventListener('url', this.handleOpenURL)
-    }
+    Linking.getInitialURL().then(url => {
+      if (url) {
+        this.doDeepLink(url)
+      }
+      // this.navigate(url);
+    }).catch(err => console.log('error occurred ', err))
+    Linking.addEventListener('url', this.handleOpenURL)
   }
-  handleOpenURL = (event: Object) => {
-    // this.props.urlRecived(event.url)
-    const splitArray = event.url.split('recovery?token=')
+  doDeepLink (url: string) {
+    const splitArray = url.split('recovery?token=')
```

Comment:
This should use a URL handler which parses out the different URI components. There's no guarantee that `token=` will be at the beginning of the URI

Reply:
@paullinator not sure what you mean by uri handler. ... based on your comments I was thinking that I could split on token= instead of recovery?token= since token = would be the spot where the token var is in the url. .. I could also add a second split on a & if there is another variable after the token.. however for this use case where we are getting the uri, we can guarantee the format since we create the link....